src/concrete/ml/torch/hybrid_model.py

```diff
@@ -223,6 +223,16 @@ def _apply(self, fn, recurse=True):
         """
         return self
 
+    def _ensure_module_on_device(self, x: torch.Tensor) -> None:
+        """Ensure the private module is on the same device as the input tensor.
+
+        Args:
+            x (torch.Tensor): The input tensor to match device with.
+        """
+        assert self.private_module is not None
+        if next(self.private_module.parameters()).device != x.device:
```

Comment:
can't there be multiple `parameters` in the `private_module`, why not iterate ?

Reply:
Yes better check if any of the param is not on the same device.